app/profile/test/test.controller.js

```diff
@@ -28,33 +31,61 @@
             testResolve.$promise.then((res) => {
                 if (res.success) {
                     vm.test = res.data;
+                    setStartButton();
+                }
+            });
+        }
+
+        function setStartButton() {
+            if (vm.test.taskList.length != 0) {
+                vm.startBtnDisabled = false;
+            }
+            userTestsResolve.$promise.then((res) => {
+                let userTest = res.data.find(val => val.test.id == vm.test.id);
+                if (userTest) {
+                    let cons = Constants.ACHIEVEMENTS_STATES;
+                    if (cons[userTest.test.achievementState] != cons.NOT_STARTED) {
+                        vm.showStopBtn = true;
+                    }
                 }
             });
         }
 
         function startTest() {
-            ProfileTestService.startTest({"testId": vm.test.id}).$promise.then(() => {
+            ProfileTestService.startTest({ "testId": vm.test.id }).$promise.then(() => {
                 $location.path('/profile/tests');
             });
         }
 
+        function stopTest() {
+            userTestsResolve.$promise.then((res) => {
+                let userTest = res.data.find(val => val.test.id == vm.test.id);
+                if (userTest) {
+                    ProfileTestService.stopTest({ 'id': userTest.id }).$promise.then(() => {
+                        $location.path('/profile/tests');
```

Comment:
Just reload page.
Also add $log('Test started successfully.'); and notification with the same message.